client/components/domains/register-domain-step/index.jsx

```diff
@@ -441,7 +441,9 @@ const RegisterDomainStep = React.createClass( {
 		const { lastDomainSearched, lastDomainStatus } = this.state,
 			matchesSearchedDomain = ( suggestion ) => ( suggestion.domain_name === lastDomainSearched ),
 			availableDomain = lastDomainStatus === domainAvailability.AVAILABLE && find( this.state.searchResults, matchesSearchedDomain ),
-			onAddMapping = ( domain ) => this.props.onAddMapping( domain, this.state );
+			onAddMapping = ( domain ) => this.props.onAddMapping( domain, this.state ),
+			exactMatchBeforeTld = ( suggestion ) => ( startsWith( suggestion.domain_name, `${ lastDomainSearched }.` ) ),
```

Comment:
We always set the `availableDomain` as recommended if returned. But yeah, wouldn't work in that case.